pkg/server/server.go

```diff
@@ -159,12 +167,65 @@ func createEnvironment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	//space delimited annotation of valid hostnames
+	var hostsList bytes.Buffer
+
+	//Loop through slice of HostNames
+	for index, value := range tempJSON.HostNames {
+		//Verify each Hostname matches regex
+		validIP := validIPAddressRegex.MatchString(value)
```

Comment:
I don't think we should always validate both hostname and IP.  We could do it on an as-needed basis by checking one and then the other if the first didn't match.  Here is an example: https://github.com/30x/k8s-pods-ingress/blob/master/ingress/pods.go#L90 _(I'm not saying it has to be this way, just saying we're clearly doing extra work that we could avoid, although I'm not sure that work is actually heavy enough to be concerned.)_